### DISCUSSION TITLE
How can I have different settings (such as allowing private access) for my tests folder?

### DISCUSSION THREAD

**Discussion by travis-leith** [microsoft/pyright#8193] [Q&A] [answered]
I find that my development process is much simpler if I am allowed to write unit tests which can access private things. I do not want to expose these things in downstream usage, but I do want to verify at a low level that each component is working as expected.
I asked copilot about how to do this and was given the following suggestion

```
"executionEnvironments": [
    {
      "root": "src",
      "reportPrivateUsage": true
    },
    {
      "root": "tests",
      "reportPrivateUsage": false
    }
  ]
```

Which does not work. The "reportPrivateUsage" setting is completely ignored. Is there some other way of achieving this?

**Comment by erictraut** [comment, marked as answer]
The approach suggested by copilot should work fine as long as you're on an up-to-date version of pyright. The support for this was added just a few weeks ago and can be found in the latest published version.

**Reply by travis-leith** [reply]
This worked after upgrading. Thanks!
